### PR TITLE
WIP #2539 bis

### PR DESCRIPTION
**Pull request by phpgeekfr**

### CODE REVIEW

File: apps/backend-functions/src/orgs.ts [hunk 1]
```diff
@@ -73,6 +76,25 @@ async function notifyOnOrgMemberChanges(before: OrganizationDocument, after: Org
   }
 }
 
+/** Checks if new org admin updated app access (possible only when org.status === 'pending' for a standard user ) */
+function hasOrgAppAccessChanged(before: OrganizationDocument, after: OrganizationDocument): boolean {
+  if (!!after.appAccess && before.status === 'pending' && after.status === 'pending') {
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/orgs.ts:81]: Why do you check is status of both is pending ? If one is accepted for example and not the other what should happen ?

File: apps/backend-functions/src/orgs.ts [hunk 2]
```diff
@@ -73,6 +76,25 @@ async function notifyOnOrgMemberChanges(before: OrganizationDocument, after: Org
...
+    for (const a of app) {
+      if (after.appAccess[a].dashboard === true && (!before.appAccess[a] || before.appAccess[a].dashboard === false)) { return true; }
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/orgs.ts:83]: Create a function with a good naming : 
```typescript
const accessChanges = (module: 'dashboard' | 'marketplace') => {
  return after.appAccess[a][module] === true && (!before.appAccess[a] || before.appAccess[a][module]  === false)
}
if (accessChanges('dashboard') || accessChanges('marketplace')) {
  return true;
}
```

File: apps/backend-functions/src/templates/mail.ts [hunk 3]
```diff
@@ -149,40 +149,37 @@ export function requestToAttendEventFromUser(fromUserFirstname: string, fromUser
...
+/**
+ * @param orgId
+ */
 const organizationCreatedTemplate = (orgId: string) =>
   `
   A new organization was created on the blockframes project,
 
-  Visit ${appUrl}${ADMIN_ACCEPT_ORG_PATH}/${orgId} to enable it.
-  `; // @TODO (#2539) => redirect to admin panel
+  Visit ${appUrl}${ADMIN_ACCEPT_ORG_PATH}/${orgId} or go to ${ADMIN_ACCEPT_ORG_PATH}/${orgId} to view it.
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/templates/mail.ts:163]: why not adding `${appUrl}` in front for the path ?

File: apps/backend-functions/src/templates/mail.ts [hunk 4]
```diff
@@ -193,17 +190,15 @@ export function organizationCreated(orgId: string): EmailRequest {
...
- * It sends and email to admin to accept or reject the request
+ * It sends an email to admin to accept or reject the request
  */
-/*export function organizationRequestedAccessToApp(orgId: string, appId: string): EmailRequest {
+export function organizationRequestedAccessToApp(orgId: string): EmailRequest {
   return {
     to: adminEmail,
     subject: 'An organization requested access to an app',
-    text: organizationRequestAccessToAppTemplate(orgId, appId)
+    text: organizationRequestAccessToAppTemplate(orgId)
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/templates/mail.ts:199]: Could we do something like in the frontend later ? `request(orgId).from('org').to('accessApp')` or something like that

File: libs/organization/src/lib/organization/+state/organization.service.ts [hunk 5]
```diff
@@ -107,32 +109,13 @@ export class OrganizationService extends CollectionService<OrganizationState> {
...
-    });
-  }*/
-
-
   public async setBlockchainFeature(value: boolean) {
     const orgId = this.query.getActiveId();
     return this.update(orgId, { isBlockchainEnabled: value });
   }
 
+  public async notifyAppAccessChange(orgId: string){
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/organization/+state/organization.service.ts:117]: yu don't need async here

File: libs/utils/src/lib/apps.ts [hunk 6]
```diff
@@ -45,4 +37,14 @@ export function createInnerAppAccess(innerAppAccess: Partial<InnerAppAccess> = {
     marketplace: false,
     ...innerAppAccess
   }
+}
+
+export function setAppName(slug: App) {
+  switch (slug) {
+    case 'catalog':
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/apps.ts:44]: Put catalog with default at the end.

File: libs/utils/src/lib/apps.ts [hunk 7]
```diff
@@ -45,4 +37,14 @@ export function createInnerAppAccess(innerAppAccess: Partial<InnerAppAccess> = {
     marketplace: false,
     ...innerAppAccess
   }
+}
+
+export function setAppName(slug: App) {
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/apps.ts:42]: It looks like a getter, not a setter.
Why not using a map ?
```typescript
const appName = {
  catalog: 'Archipel Content',
  ...
};
```

File: libs/utils/src/lib/apps.ts [hunk 8]
```diff
@@ -7,17 +7,17 @@ import { RouterQuery } from "@datorama/akita-ng-router-store";
...
 
-export interface InnerAppAccess { marketplace: boolean, dashboard: boolean };
-export type OrgAppAccess = Record<App, InnerAppAccess>;
-export type MovieAppAccess = Record<App, boolean>;
+export const module = ['dashboard', 'marketplace'] as const;
+export type Module = typeof module[number];
+
+const appName = {
+  catalog: 'Archipel Content',
+  festival: ''
```
Comment (Mussuranae) [inline libs/utils/src/lib/apps.ts:15]: `festival: 'Archipel Market'` ?

File: apps/backend-functions/src/templates/mail.ts (hunk 4 above)
Comment (phpgeekfr) [reply apps/backend-functions/src/templates/mail.ts:199]: Yes we could, but right now, org can change it's app access at creation only

**Pull request merged**